### ISSUE TITLE
conflicting `/nix-support/setup-hook` from multiple packages

### ISSUE THREAD

**Issue by blaggacao** [nlewo/nix2container#24]
When I specify two packages that both implement `/nix-support/setup-hook` in `contents = [nixpkgs.iana-etc nixpkgs.cacert];`, I eventually result with this error:

```console
       > The file /nix-support/setup-hook overrides a file with different attributes (previous: &tar.Header{Typeflag:0x30, Name:"/nix-support/setup-hook", Linkname:"", Size:396, Mode:292, Uid:0, Gid:0, Uname:"root", Gname:"root", ModTime:time.Time{wall:0x0, ext:62135596800, loc:(*time.Location)(nil)}, AccessTime:time.Time{wall:0x0, ext:62135596800, loc:(*time.Location)(nil)}, ChangeTime:time.Time{wall:0x0, ext:62135596800, loc:(*time.Location)(nil)}, Devmajor:0, Devminor:0, Xattrs:map[string]string(nil), PAXRecords:map[string]string(nil), Format:0} current: &tar.Header{Typeflag:0x30, Name:"/nix-support/setup-hook", Linkname:"", Size:200, Mode:292, Uid:0, Gid:0, Uname:"root", Gname:"root", ModTime:time.Time{wall:0x0, ext:62135596800, loc:(*time.Location)(nil)}, AccessTime:time.Time{wall:0x0, ext:62135596800, loc:(*time.Location)(nil)}, ChangeTime:time.Time{wall:0x0, ext:62135596800, loc:(*time.Location)(nil)}, Devmajor:0, Devminor:0, Xattrs:map[string]string(nil), PAXRecords:map[string]string(nil), Format:0})
       For full logs, run 'nix log /nix/store/kgp2sn9pknhp5nbf3kjryxpyv6w0djzi-layers.json.drv'.
```

... while I'm trying to figure out a work-around, I wanted to already drop a bug report in here. Maybe it's already known?

**Comment by blaggacao** [comment]
`symlinkJoin` is doing the trick. So maybe that that could be internalized into `nix2container`?

```nix
{
      contents = [
      nixpkgs.bashInteractive
      (nixpkgs.symlinkJoin {
        name = "synthetic";
        paths = [nixpkgs.iana-etc nixpkgs.cacert];
      })
    ];

}
```

**Comment by blaggacao** [comment]
`copyToRoot` :+1:

**Comment by nrdxp** [comment]
It's probably important to note that using `symlinkJoin` causes all the `paths` to be stored in a single image layer, even if you specify the layers seperately.

**Comment by nlewo** [comment]
@nrdxp i'm not sure to understand. Theorically, there are several ways to isolate `symlinkJoin` dependencies into dedicated layers.

- you can just set maxLayers on your image: in this case, Nix builds a single derivation which generates several layers (based on the [layered algo](https://grahamc.com/blog/nix-and-layered-docker-images))
- you can explicitly put a `symlinkJoin` dependency into a layer (with the `nix2container.buildImage.layers` attribute): in this case, Nix builds 2 derivations (you can benefit from the Nix cache to skip layer rebuilds) and these 2 derivations can builds at least 2 layers (one with your dependency and another one with all other `symlinkJoin` dependencies).

**Comment by nlewo** [comment]
See https://github.com/NixOS/nixpkgs/pull/179801/files

**Comment by nlewo** [comment]
Closed since #33 should help the user with this kind of issues.
